src/neo/SmartContract/ApplicationEngine.Binary.cs

```diff
@@ -23,6 +28,42 @@ protected internal StackItem BinaryDeserialize(byte[] data)
             return BinarySerializer.Deserialize(data, Limits.MaxStackSize, Limits.MaxItemSize, ReferenceCounter);
         }
 
+        protected internal string Itoa(BigInteger value, int @base)
+        {
+            return @base switch
+            {
+                10 => value.ToString(),
+                16 => value.ToString("x2"),
+                _ => throw new ArgumentException(nameof(@base)),
+            };
+        }
+
+        protected internal BigInteger Atoi(string value, int @base)
+        {
+            switch (@base)
+            {
+                case 10:
+                    {
+                        if (BigInteger.TryParse(value, out var ret))
+                        {
+                            return ret;
+                        }
+                        break;
+                    }
+                case 16:
+                    {
+                        if (BigInteger.TryParse(value, NumberStyles.HexNumber, CultureInfo.InvariantCulture, out var ret))
+                        {
+                            return ret;
+                        }
+                        break;
+                    }
+                default: throw new ArgumentException(nameof(@base));
+            }
+
+            throw new FormatException();
```

Comment:
Since we want to throw an exception, we might as well use `Parse()` directly.

Reply:
Done